src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorAchDataProcessingIntoKfsServiceImpl.java

```diff
@@ -0,0 +1,200 @@
+        } finally {
+            GlobalVariables.getMessageMap().clearErrorMessages();
+        }
+        return documentRouted;
+    }
+    
+    private void captureKfsPayeeAchProcessingErrorsForVendor(PaymentWorksVendor pmwVendorWithFailure, PaymentWorksNewVendorPayeeAchBatchReportData reportData, List<String> validationErrors) {
+        reportData.getRecordsWithProcessingErrorsSummary().incrementRecordCount();
+        reportData.addRecordWithProcessingErrors(getPaymentWorksNewVendorPayeeAchReportService().createBatchReportVendorItem(pmwVendorWithFailure, validationErrors));
+    }
+
+    private MaintenanceDocument buildPayeeAchMaintenanceDocument(KfsAchDataWrapper kfsAchDataWrapper) throws WorkflowException {
+        MaintenanceDocument paatMaintenceDoc = buildPayeeAchMaintenanceDocumentBase(kfsAchDataWrapper.getPayeeAchAccount(), kfsAchDataWrapper.getPayeeAchAccountExplanation(), 
+                                                                                    KFSConstants.MAINTENANCE_NEW_ACTION, kfsAchDataWrapper.getPayeeAchAccountNotes());
+        return paatMaintenceDoc;
+    }
+    
+    private MaintenanceDocument buildPayeeAchMaintenanceDocumentBase(PayeeACHAccount newPayeeAchAccount, String documentExplanation, String documentAction, List<Note> notesToPersist) throws WorkflowException{
+        MaintenanceDocument paatMaintenceDoc = (MaintenanceDocument) getDocumentService().getNewDocument(CUPdpConstants.PAYEE_ACH_ACCOUNT_EXTRACT_MAINT_DOC_TYPE);
+        paatMaintenceDoc.getDocumentHeader().setDocumentDescription(buildPayeeAchMaintenanceDocumentDescription(newPayeeAchAccount));
+        paatMaintenceDoc.getDocumentHeader().setExplanation(documentExplanation);
+        
+        PayeeACHAccountMaintainableImpl maintainable = (PayeeACHAccountMaintainableImpl) paatMaintenceDoc.getNewMaintainableObject();
+        maintainable.setMaintenanceAction(KFSConstants.MAINTENANCE_NEW_ACTION);
+        PayeeACHAccount achAccount = (PayeeACHAccount) maintainable.getDataObject();
+        
+        achAccount.setPayeeIdentifierTypeCode(newPayeeAchAccount.getPayeeIdentifierTypeCode());
+        achAccount.setPayeeIdNumber(newPayeeAchAccount.getPayeeIdNumber());
+        achAccount.setBankRoutingNumber(newPayeeAchAccount.getBankRoutingNumber());
+        achAccount.setBankAccountNumber(newPayeeAchAccount.getBankAccountNumber());
+        achAccount.setBankAccountTypeCode(newPayeeAchAccount.getBankAccountTypeCode());
+        achAccount.setPayeeName(newPayeeAchAccount.getPayeeName());
+        achAccount.setPayeeEmailAddress(newPayeeAchAccount.getPayeeEmailAddress());
+        achAccount.setAchTransactionType(newPayeeAchAccount.getAchTransactionType());
+        achAccount.setActive(newPayeeAchAccount.isActive());
+        
+        for (Note note : notesToPersist) {
+            note.setRemoteObjectIdentifier(paatMaintenceDoc.getObjectId());
```

Comment:
I do not understand why I would do that kind of check here when it is not done in any of the other locations where code similar to this exists.  What should be done if that check were performed?

Reply:
My concern is that I'm not sure if the document has an object ID at this point, unless it has been saved. Also, if you're associating the notes with the document instead of the BO, you should double-check whether the notes should be associated with the document header object instead of the document itself.

Alternatively, I believe the DocumentService will set up the remote identifiers appropriately when saving or submitting the document.

Reply:
As per further discussion with Nancy, the setup from this line should get overridden accordingly when the document gets saved/submitted, so it shouldn't cause problems with the PAAT documents. However, it would still be a good idea to clean up this line at some point if it is not needed.